### PR TITLE
Add VerificationMode to ValidatePeerCertificateChainReq SessionReq

### PR DESCRIPTION
**Pull request by rmehta19**
Add VerificationMode to ValidatePeerCertificateChainReq SessionReq.

### CODE REVIEW

File: internal/v2/s2av2.go [hunk 1]
```diff
@@ -29,11 +29,12 @@ type s2av2TransportCreds struct {
 	serverName string
 	s2av2Address string
 	tokenManager *tokenmanager.AccessTokenManager
+	verificationMode s2av2pb.ValidatePeerCertificateChainReq_VerificationMode
 }
 
 // NewClientCreds returns a client-side transport credentials object that uses
 // the S2Av2 to establish a secure connection with a server.
-func NewClientCreds(s2av2Address string) (credentials.TransportCredentials, error) {
+func NewClientCreds(s2av2Address string, verificationMode s2av2pb.ValidatePeerCertificateChainReq_VerificationMode) (credentials.TransportCredentials, error) {
```
Comment (matthewstevenson88) [inline internal/v2/s2av2.go:37]: We don't want to expose the proto directly to the users, so I think the right way to do this is:

1. Add a "verification mode" enum to the options file: https://github.com/google/s2a-go/blob/main/s2a_options.go

2. Add a "verification mode" field to each of the client/server options, which defaults to the "connecting to Google" value

3. Plumb this value through to cert_verifier.go, and then convert to the proto enum when preparing the SessionReq message.

Does that sound reasonable?

Comment (rmehta19) [reply internal/v2/s2av2.go:37]: Thanks for pointing this out! In this PR, I added a verificationMode field to the client and server options in s2a_options, but I did expose the proto, I'll change it to the enum.

File: s2a_options.go [hunk 2]
```diff
@@ -64,6 +64,14 @@ func NewUID(name string) Identity {
 	return &uid{uid: name}
 }
 
+type VerificationModeType int
```
Comment (matthewstevenson88) [inline s2a_options.go:67]: Please add a top-level comment for this, e.g. "VerificationModeType specifies the mode that S2A must use to verify the peer certificate chain."

File: s2a_options.go [hunk 3]
```diff
@@ -105,11 +113,16 @@ type ClientOptions struct {
 	EnsureProcessSessionTickets *sync.WaitGroup
 	// If true, enables the use of S2Av2.
 	EnableV2 bool
+	// VerificationMode specifies the verification mode supported by S2Av2.
```
Comment (matthewstevenson88) [inline s2a_options.go:116]: Suggested edit: VerificationMode specifies the mode that S2A must use to verify the peer certificate chain.

File: s2a_options.go [hunk 4]
```diff
@@ -123,11 +136,16 @@ type ServerOptions struct {
 	S2AAddress string
 	// If true, enables the use of S2Av2.
 	EnableV2 bool
+	// VerificationMode specifies the verification mode supported by S2Av2.
```
Comment (matthewstevenson88) [inline s2a_options.go:139]: Suggested edit: VerificationMode specifies the mode that S2A must use to verify the peer certificate chain.

**Review by matthewstevenson88 — APPROVED** [review]
LGTM with 3 nits.

File: s2a_options.go (hunk 2 above)
Comment (rmehta19) [reply s2a_options.go:67]: Will do, thanks!

File: s2a_options.go (hunk 3 above)
Comment (rmehta19) [reply s2a_options.go:116]: Will do, thanks!

File: s2a_options.go (hunk 4 above)
Comment (rmehta19) [reply s2a_options.go:139]: Will do, thanks!

**Pull request merged**